Foundation/NSTimeZone.swift

```diff
@@ -29,6 +29,26 @@ open class NSTimeZone : NSObject, NSCopying, NSSecureCoding, NSCoding {
     }
 
     public init?(name tzName: String, data aData: Data?) {
+#if os(Android)
+        var tzName = tzName
+        if tzName == "UTC" || tzName == "GMT" {
+            tzName = "GMT+0000"
+        }
+        else {
+            var now = time(nil), info = tm()
+            if localtime_r(&now, &info) != nil &&
+                (tzName == "System" || tzName == String(cString: info.tm_zone)) {
+                (info.tm_gmtoff < 0 ? "-" : "+").withCString {
+                    tzName = String(format: "GMT%s%02d%02d",
```

Comment:
This is a valiant effort but very much incorrect. An `NSTimeZone` instance for a particular time zone will have different offsets from GMT depending on the date and time; it is not equivalent to a fixed GMT offset. When you don't have the information to instantiate the requested time zone, the correct result can only be `nil`.